### PR TITLE
New "Get days in month" goodie

### PR DESCRIPTION
**Pull request by 0xbsec**
Complete of the aborded pull request #812 
numberdaysmonth

### CODE REVIEW

**Comment by daxtheduck** [comment]
@0xbsec Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/NumberDaysMonth.pm [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+# Caching - http://docs.duckduckhack.com/backend-reference/api-reference.html#caching`
+zci is_cached => 1;
+
+# Triggers - http://docs.duckduckhack.com/walkthroughs/calculation.html#triggers
+triggers any => 'how many days are in', 'what is the number of days in', 'number of days in';
+
+# Handle statement
+handle remainder => sub {
+    my $remainder = $_;
+	return unless $remainder =~ qr/^\s*\w+\s*$/;
```
Comment (moollaza) [inline lib/DDG/Goodie/NumberDaysMonth.pm]: @0xbsec can you fix the indentation throughout the file? We use 4-space soft tabs :+1:

File: lib/DDG/Goodie/NumberDaysMonth.pm [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        "jul" => 7,
+        "aug" => 8,
+        "sep" => 9,
+        "oct" => 10,
+        "nov" => 11,
+        "dec" => 12
+		);
+	}else{
+		%months = (
+        "january" =>     1,
```
Comment (moollaza) [inline lib/DDG/Goodie/NumberDaysMonth.pm]: We shouldn’t need this hash. We can simply take the first 3 letters of the string and see if it exists in the first hash :+1:

File: lib/DDG/Goodie/NumberDaysMonth.pm (hunk 1 above)
Comment (moollaza) [inline lib/DDG/Goodie/NumberDaysMonth.pm]: we might as well use the keys in our months hash to match here. You could construct a regex or use `grep` or similar.

File: lib/DDG/Goodie/NumberDaysMonth.pm [hunk 3]
```diff
@@ -0,0 +1,82 @@
+package DDG::Goodie::NumberDaysMonth;
```
Comment (moollaza) [inline lib/DDG/Goodie/NumberDaysMonth.pm]: I think a package name like `DaysInMonth` would be a little clearer?

**Comment by 0xbsec** [comment]
@moollaza Please check now. Your are right in changing the name, Changing the file names and the package name will do it ?

**Comment by 0xbsec** [comment]
Hi @moollaza , any news on this ?

**Pull request closed without merging**